core/attestation.go

```diff
@@ -51,8 +51,19 @@ func (p *G1Point) Serialize() []byte {
 	return bn254utils.SerializeG1(p.G1Affine)
 }
 
-func (p *G1Point) Deserialize(data []byte) *G1Point {
-	return &G1Point{bn254utils.DeserializeG1(data)}
+func (p *G1Point) Deserialize(data []byte) (*G1Point, error) {
+	point, err := bn254utils.DeserializeG1(data)
+	if err != nil {
+		return nil, err
+	}
+	return &G1Point{point}, nil
+}
+
+func (p *G1Point) Copy() *G1Point {
```

Comment:
Will it make more sense to name it "Clone"? This is deep copying (creating a new instance)

Reply:
Yes, this is a deep copy. Is `Clone` a standard usage for this? Would `DeepCopy` be clearest?

Reply:
Yea, clone is usually deep copy